src/main/java/com/amazon/dlic/auth/http/jwt/HTTPExtensionJwtAuthenticator.java

```diff
@@ -0,0 +1,274 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ *
+ * Modifications Copyright OpenSearch Contributors. See
+ * GitHub history for details.
+ */
+
+package com.amazon.dlic.auth.http.jwt;
+
+import java.security.AccessController;
+import java.security.Key;
+import java.security.KeyFactory;
+import java.security.NoSuchAlgorithmException;
+import java.security.PrivilegedAction;
+import java.security.PublicKey;
+import java.security.spec.InvalidKeySpecException;
+import java.security.spec.X509EncodedKeySpec;
+import java.util.Arrays;
+import java.util.Map.Entry;
+import java.util.regex.Pattern;
+
+import io.jsonwebtoken.Claims;
+import io.jsonwebtoken.JwtParser;
+import io.jsonwebtoken.Jwts;
+import io.jsonwebtoken.io.Decoders;
+import io.jsonwebtoken.security.WeakKeyException;
+import org.apache.commons.lang3.RandomStringUtils;
+import org.apache.hc.core5.http.HttpHeaders;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+import org.greenrobot.eventbus.Subscribe;
+import org.opensearch.OpenSearchSecurityException;
+import org.opensearch.SpecialPermission;
+import org.opensearch.common.util.concurrent.ThreadContext;
+import org.opensearch.rest.BytesRestResponse;
+import org.opensearch.rest.RestChannel;
+import org.opensearch.rest.RestRequest;
+import org.opensearch.rest.RestStatus;
+import org.opensearch.security.auth.HTTPAuthenticator;
+import org.opensearch.security.authtoken.jwt.EncryptionDecryptionUtil;
+import org.opensearch.security.securityconf.DynamicConfigModel;
+import org.opensearch.security.user.AuthCredentials;
+
+public class HTTPExtensionJwtAuthenticator implements HTTPAuthenticator {
+
+    protected final Logger log = LogManager.getLogger(this.getClass());
+
+    private static final Pattern BASIC = Pattern.compile("^\\s*Basic\\s.*", Pattern.CASE_INSENSITIVE);
+    private static final String BEARER = "bearer ";
+
+    //TODO: TO SEE IF WE NEED THE FINAL FOR FOLLOWING
+    private JwtParser jwtParser;
+    //private final boolean isDefaultAuthHeader;
+    //private final String rolesKey;
+    private String subjectKey;
+    private String requireAudience;
+    private String requireIssuer;
+    //private final String httpheader;
+
+    private String signingKey;
+    private String encryptionKey;
+
+    public HTTPExtensionJwtAuthenticator() {
+        super();
+        init();
+    }
+
+    // FOR TESTING
+    public HTTPExtensionJwtAuthenticator(String signingKey, String encryptionKey){
+        this.signingKey = signingKey;
+        this.encryptionKey = encryptionKey;
+        init();
+    }
+
+    private void init() {
+        JwtParser _jwtParser = null;
+
+        try {
+            if(signingKey == null || signingKey.length() == 0) {
```

Comment:
Should this throw an error instead of just logging?

Reply:
Yes this will directly skip the `else` and throw another error log L110 with an runtime exception at [L112](https://github.com/opensearch-project/security/pull/2672/files#diff-efd217975f23af4bb84808e2771bfc85211f12b59bb619da430a3363902207ceR112). The way we want to do it in this way is because we would like to log some specific information for user to know the reason of failure creation of this token. There are also some `try`/`catch` statement in the `else` for checking encryption algorithm, and each of them will work in the same pattern. It will tell the user why they failed first by these error log and give them the runtime exception error at L112.